File: src/geometryService/cpp/stractNet/ByteArrayFrag.cpp
///////////////////////////////////////////////////////////
//  ByteArrayFrag.cpp
//  Implementation of the Class ByteArrayFrag
//  Created on:      06-Aug-2008 7:52:41 AM
//  Original author: david.h.loman
///////////////////////////////////////////////////////////

#include "ByteArrayFrag.h"
///////////////////////////////////////////////////////////////////////
/*
 *
 * xSTRUCTORS
 *
 */
///////////////////////////////////////////////////////////////////////

ByteArrayFrag::~ByteArrayFrag(){}
ByteArrayFrag::ByteArrayFrag(byte b[]){
	MsgFrag("ByteArrayFrag", b, MsgFrag::mftByteArray);
}
ByteArrayFrag::ByteArrayFrag(DataInputStream& in){
	MsgFrag("ByteArrayFrag", in, MsgFrag::mftByteArray);
}

///////////////////////////////////////////////////////////////////////
/*
 *
 * METHODS
 *
 */
///////////////////////////////////////////////////////////////////////

void ByteArrayFrag::Deserialize(DataInputStream& in){
	try {
		//Read length of array
		int len = in.readInt();

		//Assign array
		this->field = byte[len];

		//Read byte[]
		in.read(this->field );

	} catch (IOException ioe) {
		SNRoot::err("IOException in Deserialize()", 0);
	}
	return ;
}


void ByteArrayFrag::Serialize_Specific(DataOutputStream& out){
	try {
		//Write length of array
		out.writeInt(this->field.length);

		//Write byte[]
		out.write(this->field);

	} catch (IOException ioe) {
		SNRoot::err("IOException in Serialize()", 0);
	}

	return;
}

File: src/superceded_GS/java/stractNet/src/stractNet/messaging/msgs/frags/ByteArrayFrag.java
package stractNet.messaging.msgs.frags;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;

public class ByteArrayFrag extends MsgFrag<byte[]> {
	public ByteArrayFrag(byte[] b) {
		super("ByteArrayFrag", b, MsgFrag.mftByteArray);
	}

	public ByteArrayFrag(DataInputStream in) {
		super("ByteArrayFrag", in, MsgFrag.mftByteArray);
	}

	@Override
	public void Deserialize(DataInputStream in) {
		try {
			//Read length of array
			int len = in.readInt();
			
			//Assign array
			this.field = new byte[len];
			
			//Read byte[]
			in.read(this.field );

		} catch (IOException ioe) {
			this.err("IOException in Deserialize()", 0);
		}
		return ;
	}

	@Override
	protected void Serialize_Specific(DataOutputStream out) {
		try {
			//Write length of array
			out.writeInt(this.field.length);
						
			//Write byte[]
			out.write(this.field);
			
		} catch (IOException ioe) {
			this.err("IOException in Serialize()", 0);
		}

		return;
	}

}
